sysdeps/sigma/generic/generic.cpp

```diff
@@ -0,0 +1,110 @@
+#include <mlibc/sysdeps.hpp>
+#include <bits/ensure.h>
+#include <mlibc/debug.hpp>
+#include <errno.h>
+#include <dirent.h>
+#include <fcntl.h>
+
+#include <libsigma/thread.h>
+
+
+#define STUB_ONLY { __ensure(!"STUB_ONLY function was called"); __builtin_unreachable(); }
+
+namespace mlibc {
+    // Unknown Functions
+    int sys_tcb_set(void *pointer){
+	    libsigma_set_fsbase(reinterpret_cast<uint64_t>(pointer));
+        return 0;
+    }
+    
+    int sys_sigaction(int signum, const struct sigaction *act, struct sigaction *oldact) STUB_ONLY
+    
+    // Memory Related Functions
+    int sys_vm_map(void *hint, size_t size, int prot, int flags, int fd, off_t offset, void **window) STUB_ONLY
+    #ifndef MLIBC_BUILDING_RTDL
+    int sys_vm_remap(void *pointer, size_t size, size_t new_size, void **window) STUB_ONLY
+    #endif
+    int sys_vm_unmap(void *pointer, size_t size) STUB_ONLY
+
+    // Misc functions
+    #ifndef MLIBC_BUILDING_RTDL
+    int sys_clock_get(int clock, time_t *secs, long *nanos) STUB_ONLY
+    int sys_futex_wait(int *pointer, int expected) STUB_ONLY
+    int sys_futex_wake(int *pointer) STUB_ONLY
+    #endif
+
+    // Task functions
+    #ifndef MLIBC_BUILDING_RTDL
+    void sys_exit(int status){
+        (void)(status);
+        libsigma_kill();
+    }
+
+    int sys_sleep(time_t *secs, long *nanos) STUB_ONLY
+    int sys_fork(pid_t *child) STUB_ONLY
+    int sys_execve(const char *path, char *const argv[], char *const envp[]) STUB_ONLY
+    int sys_kill(pid_t pid, int signal) STUB_ONLY
+    int sys_waitpid(pid_t pid, int *status, int flags, pid_t *ret_pid) STUB_ONLY
+    int sys_getrusage(int who, struct rusage *usage) STUB_ONLY
+    int sys_sigprocmask(int how, const sigset_t *__restrict set, sigset_t *__restrict retrieve) STUB_ONLY
+    void sys_yield() STUB_ONLY
+    gid_t sys_getgid() STUB_ONLY
+    gid_t sys_getegid() STUB_ONLY
+    uid_t sys_getuid() STUB_ONLY
+    uid_t sys_geteuid() STUB_ONLY
+    pid_t sys_getpid() STUB_ONLY
+    pid_t sys_getppid() STUB_ONLY
+    #endif
```

Comment:
Note that implementing all sysdeps functions will disable some graceful fallbacks in mlibc. For example, may functions will log an error and fail with ENOSYS if the sysdeps function is unavailable. This makes mlibc panic instead.

Reply:
Do all function do this?

Reply:
Not all iirc, some assume the sysdep function is present (`mkstemp` assumes `sys_open` exists, `mkdtemp` assumes `sys_mkdir` exists, for example)

Reply:
Is there a list somewhere, because going through all calls to all sysdep functions seems very tedious.

Reply:
Not all functions have graceful fallbacks but all functions should either panic or fallback to an error return. If some functions use `[[gnu::weak]]` sysdeps without checking for them, that's a bug.